repository/src/main/java/org/alfresco/repo/event2/EventGenerator.java

```diff
@@ -392,36 +425,41 @@ protected void onShutdown(ApplicationEvent applicationEvent)
         @Override
         public void afterCommit()
         {
-            try
+            if(isTransactionCommitted())
             {
-                final Consolidators consolidators = getTxnConsolidators(this);
-
-                // Node events
-                for (Map.Entry<NodeRef, EventConsolidator> entry : consolidators.getNodes().entrySet())
+                try
                 {
-                    EventConsolidator eventConsolidator = entry.getValue();
-                    sendEvent(entry.getKey(), eventConsolidator);
+                    final Consolidators consolidators = getTxnConsolidators(this);
+
+                    // Node events
+                    for (Map.Entry<NodeRef, EventConsolidator> entry : consolidators.getNodes().entrySet())
+                    {
+                        EventConsolidator eventConsolidator = entry.getValue();
+                        sendEvent(entry.getKey(), eventConsolidator);
+                    }
+
+                    // Child assoc events
+                    for (Map.Entry<ChildAssociationRef, ChildAssociationEventConsolidator> entry : consolidators.getChildAssocs()
+                                                                                                           .entrySet())
+                    {
+                        ChildAssociationEventConsolidator eventConsolidator = entry.getValue();
+                        sendEvent(entry.getKey(), eventConsolidator);
+                    }
+
+                    // Peer assoc events
+                    for (Map.Entry<AssociationRef, PeerAssociationEventConsolidator> entry : consolidators.getPeerAssocs()
+                                                                                                     .entrySet())
```

Comment:
This formatting looks very odd, please move everything in one line.

Reply:
done